### PR TITLE
ref: Split out the DDL parts of the table schema class.

### PR DESCRIPTION
**Pull request by lynnagara**
Simplifies the base TableSchema class by removing the DDL related parts.
Also updates the querylog and events_ro schemas to use the simplified
TableSchema class directly since these tables are not being created by
the current migration system. Once the old migration system is not in
use all of the table and engine specific schema classes can be replaced
by this.

### CODE REVIEW

**Review by fpacifici — COMMENTED** [review]
> Once the old migration system is not in use all of the table and engine specific schema classes can be replaced by this.

Just to confirm this. This means that our table definition, storage definition and schemas will not have DDL details in them. They will only be in the migration code, which is not connected to a storage but only to a storage set. Is this correct ?

If yes, how are we going to deal, in the long term, with the attributes that are shared between DDL operations and TableSchema? I am thinking about the columnset itself, the table name, and potentially the content of the order_by (which we could use for query processor to decide when to promote something to prewhere). Are we going to duplicate them or will there be a way to share these attributes ?

File: snuba/datasets/storages/querylog.py [hunk 1]
```diff
@@ -60,14 +60,11 @@
     ]
 )
 
-schema = MergeTreeSchema(
+schema = WritableTableSchema(
```
[the remark this answers is not in the document]
Comment (lynnagara) [reply snuba/datasets/storages/querylog.py:63]: Yeah that's right. The querylog storage was never added to the list of storages to migrate so no one should be running the migrations for it right now. https://github.com/getsentry/snuba/blob/master/snuba/migrations/migrate.py#L16-L26.

**Comment by lynnagara** [comment]
The plan is to maintain the duplicate list of columns and the table names. I don't think they will ever be shared since they are the same to start by can diverge since the ones in the migration never change and the schema only represents the latest state. It's possible that one can autogenerate the other but I think that's hard to support all cases, like if raw SQL is provided. It also seems to me that it's possible that all of the column details like types, defaults, codecs, etc might not be useful for query processing, just the column names are really used.

File: snuba/datasets/storages/querylog.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fpacifici) [reply snuba/datasets/storages/querylog.py:63]: ok, mind adding a comment on the storage explaining that reasoning (as long as we do not start running the new system).

**Review by fpacifici — APPROVED** [review]
good point.

Please just see my comment inline

**Pull request merged**